does not compile

`quicer/c_src/quicer_config.c:324:37: error: passing argument 3 of ‘enif_get_uint64’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   if (!enif_get_uint64(env, evalue, &value0))
/home/pi/erl/24.0.2/usr/include/erl_nif_api_funcs.h:135:89: note: expected ‘ErlNifUInt64 *’ {aka ‘long long unsigned int *’} but argument is of type ‘long unsigned int *’
 ERL_NIF_API_FUNC_DECL(int,enif_get_uint64,(ErlNifEnv*, ERL_NIF_TERM term, ErlNifUInt64* ip));
                                                                           ~~~~~~~~~~~~~~^~
/home/pi/erl/24.0.2/usr/include/erl_nif.h:345:76: note: in definition of macro ‘ERL_NIF_API_FUNC_DECL’
 #  define ERL_NIF_API_FUNC_DECL(RET_TYPE, NAME, ARGS) extern RET_TYPE NAME ARGS
                                                                            ^~~~
cc1: all warnings being treated as errors
make[3]: *** [CMakeFiles/quicer_static.dir/build.make:76: CMakeFiles/quicer_static.dir/c_src/quicer_config.c.o] Error 1
`
erlang 24.0.2
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -60,3 +60,36 @@ jobs:
           name: linux-ct-logs
           path: _build/test/
           retention-days: 5
+
+  multiarch:
+    ### note:
+    ### -  only verify compilation because qemu user env has no support for some socket opt
+    runs-on: ubuntu-latest
+    steps:
+    - run: |
+        docker run --rm --privileged multiarch/qemu-user-static:register --reset
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: make ci arm32v7
+      uses: docker://emqx/build-env:erl24.0.1-emqx-1-alpine-arm32v7
+      with:
+        args: >
+          make
+
+    - name: make ci arm64v8
+      uses: docker://emqx/build-env:erl24.0.1-emqx-1-alpine-arm64v8
+      with:
+        args: >
+          make
+
+    - name: Archive CT Logs
+      uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: linux-ct-logs
+        path: _build/test/
+        retention-days: 5

--- a/c_src/quicer_config.c
+++ b/c_src/quicer_config.c
@@ -261,12 +261,21 @@ get_uint8_from_map(ErlNifEnv *env,
     {
       return false;
     }
+
   unsigned int value0 = 0;
+
   if (!enif_get_uint(env, evalue, &value0))
     {
       return false;
     }
+
+  if (value0 > UINT8_MAX)
+    {
+      return false;
+    }
+
   *value = (uint8_t)value0;
+
   return true;
 }
 
@@ -281,11 +290,18 @@ get_uint16_from_map(ErlNifEnv *env,
     {
       return false;
     }
+
   unsigned int value0 = 0;
   if (!enif_get_uint(env, evalue, &value0))
     {
       return false;
     }
+
+  if (value0 > UINT16_MAX)
+    {
+      return false;
+    }
+
   *value = (uint16_t)value0;
   return true;
 }
@@ -301,13 +317,8 @@ get_uint32_from_map(ErlNifEnv *env,
     {
       return false;
     }
-  unsigned long value0 = 0;
-  if (!enif_get_uint64(env, evalue, &value0))
-    {
-      return false;
-    }
-  *value = (uint32_t)value0;
-  return true;
+
+  return enif_get_uint(env, evalue, value);
 }
 
 bool
@@ -321,13 +332,12 @@ get_uint64_from_map(ErlNifEnv *env,
     {
       return false;
     }
-  unsigned long value0 = 0;
-  if (!enif_get_uint64(env, evalue, &value0))
-    {
-      return false;
-    }
-  *value = (uint64_t)value0;
-  return true;
+
+#ifdef __APPLE__
+  return enif_get_uint64(env, evalue, (unsigned long *)value);
+#else
+  return enif_get_uint64(env, evalue, value);
+#endif
 }
 
 ERL_NIF_TERM

--- a/c_src/quicer_stream.c
+++ b/c_src/quicer_stream.c
@@ -375,8 +375,6 @@ send2(ErlNifEnv *env, __unused_parm__ int argc, const ERL_NIF_TERM argv[])
   // Allocates and builds the buffer to send over the stream.
   //
 
-  // ensure type of ErlNifBinary.size
-  assert(sizeof(size_t) == sizeof(uint64_t));
   assert(bin.data != NULL);
   QUIC_BUFFER *SendBuffer
       = CXPLAT_ALLOC_NONPAGED(sizeof(QUIC_BUFFER) + bin.size, QUICER_SND_BUFF);

--- a/test/quicer_SUITE.erl
+++ b/test/quicer_SUITE.erl
@@ -68,6 +68,9 @@
         , tc_alpn/1
         , tc_alpn_mismatch/1
         , tc_idle_timeout/1
+
+        %% testcase to verify env works
+        %% , tc_network/1
         ]).
 
 -export([tc_app_echo_server/1]).
@@ -145,6 +148,10 @@ end_per_testcase(_TestCase, _Config) ->
 %%%===================================================================
 %%% Individual Test Cases (from groups() definition)
 %%%===================================================================
+%% tc_network(_Config) ->
+%%   {ok, _} = gen_tcp:listen(12456, []),
+%%   {ok, _} = gen_udp:open(12456, []).
+
 tc_nif_module_load(_Config) ->
   {module, quicer_nif} = c:l(quicer_nif).
 

